Client/Frontend/Settings/HomepageSettings/WallpaperSettings/v2/WallpaperSettingsViewController.swift

```diff
@@ -0,0 +1,285 @@
+        collectionView.dataSource = self
+        collectionView.delegate = self
+
+        collectionView.register(
+            WallpaperCollectionViewCell.self,
+            forCellWithReuseIdentifier: WallpaperCollectionViewCell.cellIdentifier)
+        collectionView.register(WallpaperSettingsHeaderView.self,
+                                forSupplementaryViewOfKind: UICollectionView.elementKindSectionHeader,
+                                withReuseIdentifier: WallpaperSettingsHeaderView.cellIdentifier)
+        return collectionView
+    }()
+
+    private lazy var activityIndicatorView: UIActivityIndicatorView = .build { view in
+        view.style = .large
+        view.isHidden = true
+    }
+
+    // MARK: - Initializers
+    init(viewModel: WallpaperSettingsViewModel,
+         notificationCenter: NotificationProtocol = NotificationCenter.default) {
+        self.viewModel = viewModel
+        self.notificationCenter = notificationCenter
+        super.init(nibName: nil, bundle: nil)
+    }
+
+    required init?(coder aDecoder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    // MARK: - View setup & lifecycle
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        setupView()
+        applyTheme()
+        setupNotifications(forObserver: self,
+                           observing: [.DisplayThemeChanged, UIContentSizeCategory.didChangeNotification])
+    }
+
+    override func viewWillAppear(_ animated: Bool) {
+        super.viewWillAppear(animated)
+        applyTheme()
+
+        collectionView.selectItem(at: viewModel.selectedIndexPath, animated: false, scrollPosition: [])
+    }
+
+    override func traitCollectionDidChange(_ previousTraitCollection: UITraitCollection?) {
+        super.traitCollectionDidChange(previousTraitCollection)
+
+        if previousTraitCollection?.horizontalSizeClass != traitCollection.horizontalSizeClass
+            || previousTraitCollection?.verticalSizeClass != traitCollection.verticalSizeClass {
+            updateOnRotation()
+        }
+    }
+
+    override func viewWillTransition(to size: CGSize, with coordinator: UIViewControllerTransitionCoordinator) {
+        super.viewWillTransition(to: size, with: coordinator)
+        updateOnRotation()
+    }
+}
+
+// MARK: - CollectionView Data Source
+extension WallpaperSettingsViewController: UICollectionViewDelegate, UICollectionViewDataSource {
+
+    func numberOfSections(in collectionView: UICollectionView) -> Int {
+        return viewModel.numberOfSections
+    }
+
+    func collectionView(_ collectionView: UICollectionView, numberOfItemsInSection section: Int) -> Int {
+        return viewModel.numberOfWallpapers(in: section)
+    }
+
+    func collectionView(_ collectionView: UICollectionView,
+                        viewForSupplementaryElementOfKind kind: String,
+                        at indexPath: IndexPath) -> UICollectionReusableView {
```

Comment:
```suggestion
    func collectionView(_ collectionView: UICollectionView,
                        viewForSupplementaryElementOfKind kind: String,
                        at indexPath: IndexPath
    ) -> UICollectionReusableView {
```
Nit: it makes identifying the return super qwack.